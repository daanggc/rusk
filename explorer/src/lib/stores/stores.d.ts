type AppStore = import("svelte/store").Readable<AppStoreContent> & {
  setTheme: (value: boolean) => void;
};

type AppStoreContent = {
  blocksListEntries: number;
  chainInfoEntries: number;
  darkMode: boolean;
  fetchInterval: number;
  isSmallScreen: boolean;
  hasTouchSupport: boolean;
  marketDataFetchInterval: number;
  statsFetchInterval: number;
  transactionsListEntries: number;
};

type MarketDataStore =
  import("svelte/store").Readable<MarketDataStoreContent> & {
    isDataStale: () => boolean;
  };

type MarketDataStoreContent = {
  data: MarketData | null;
  error: Error | null;
  isLoading: boolean;
  lastUpdate: Date | null;
};

type NetworkOption = {
  label: string;
  value: URL;
};
