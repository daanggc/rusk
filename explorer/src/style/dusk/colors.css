:root {
  /* Dark Greys */

  --granite-grey: #636167;
  --jet: #2e2d30;
  --smokey-black: #101010;

  /* Greys */

  --light-grey: #e2dfe9;
  --magnolia: #edeaf3;
  --silver-metallic: #a8a5af;
  --taupe-grey: #908e94;

  /* Blues */

  --cornflower: #71b1ff;
  --cornflower-dark: #0b79ff;
  --cornflower-light: #a4cdff;

  --accent-light-900: #04336b;
  --accent-dark-900: #abd1ff;

  /* Status */

  --success: #16db93;
  --warning: #ffcf23;
  --error: #ed254e;
  --info: #71b1ff;

  --success-500: #16db93;
  --success-700: #0f9363;
  --warning-500: #ffcf23;
  --warning-700: #d1a300;
  --error-500: #ed254e;
  --error-700: #8e112c;
  --info-500: #71b1ff;
  --info-700: #0863d1;
}
