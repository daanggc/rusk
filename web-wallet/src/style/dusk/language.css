:root {
  /* Theme colours */

  --page-background-color: var(--smokey-black);
  --background-color: var(--light-grey);
  --background-color-alt: #fff;
  --primary-color: var(--smokey-black);
  --secondary-color: var(--cornflower);
  --secondary-color-variant-dark: var(--cornflower-dark);
  --secondary-color-variant-light: var(--cornflower-light);
  --surface-color: var(--magnolia);
  --danger-color: var(--error);
  --error-color: var(--error);
  --info-color: var(--info);
  --success-color: var(--success-500);
  --success-color-variant-dark: var(--success-700);
  --warning-color: var(--warning);
  --muted-color: var(--granite-grey);

  --on-background-color: var(--smokey-black);
  --on-primary-color: var(--light-grey);
  --on-secondary-color: var(--smokey-black);
  --on-surface-color: var(--smokey-black);
  --on-danger-color: #fff;
  --on-error-color: #fff;
  --on-info-color: var(--smokey-black);
  --on-success-color: var(--smokey-black);
  --on-warning-color: var(--smokey-black);

  /* Typography */

  --main-font-family: "Soehne", "Helvetica Neue", Helvetica, Arial, sans-serif;
  --main-font-size: 16px;
  --mono-font-family: "Soehne Mono", monospace;

  /* Layout */

  --small-gap: 0.5em;
  --default-gap: 1em;
  --medium-gap: 1.25em;
  --large-gap: 2em;

  /* Grid */

  --mobile-grid-template-columns: 1fr;
  --tablet-grid-template-columns: 1fr 1fr;
  --desktop-grid-template-columns: 1fr 1fr 1fr;

  /* Controls */

  --control-border-radius-size: 1.5em;
  --control-border-size: 0.0625em;
  --control-font-size: 1em;
  --control-icon-size: 2em;
  --control-padding: 0.9em 1.25em;
  --control-small-font-size: 0.875em;
  --control-small-padding: 0.625em 0.875em;

  --non-button-control-bg-color: #fff;
  --non-button-control-text-color: var(--smokey-black);

  --control-bg-color: var(--smokey-black);
  --control-bg-color-hover: var(--cornflower);
  --control-border-color: transparent;
  --control-border-color-hover: transparent;
  --control-text-color: #fff;
  --control-text-color-hover: var(--smokey-black);

  --control-secondary-bg-color: #fff;
  --control-secondary-bg-color-hover: var(--silver-metallic);
  --control-secondary-border-color: transparent;
  --control-secondary-border-color-hover: transparent;
  --control-secondary-text-color: var(--smokey-black);

  --control-tertiary-bg-color: transparent;
  --control-tertiary-bg-color-hover: var(--cornflower);
  --control-tertiary-border-color: var(--smokey-black);
  --control-tertiary-border-color-hover: var(--cornflower);
  --control-tertiary-text-color: var(--smokey-black);

  /* Anchors */

  --anchor-color-on-surface: var(--accent-light-900);
  --anchor-color: var(--secondary-color);
  --anchor-color-visited: var(--secondary-color);
  --anchor-color-hover: var(--secondary-color-variant-light);
  --anchor-color-active: var(--secondary-color-variant-light);

  /* Progress bars */

  --progress-bar-height: 0.3125em;
  --progress-bg-color: var(--primary-color);
  --progress-filler-color: var(--success-color);

  /* Steppers */

  --stepper-bar-height: 0.3125em;
  --stepper-rail-color: var(--background-color);
  --stepper-rail-color-variant-secondary: var(--primary-color);
  --stepper-filler-color: var(--success-color);
  --stepper-on-filler-color: var(--on-success-color);
  --stepper-on-rail-color: var(--on-background-color);
  --stepper-on-rail-variant-secondary-color: var(--on-primary-color);

  /* Checkbox control */

  --checkbox-control-size: 1.2em;
  --checkbox-control-border-radius: 20%;
  --checkbox-control-border-width: 0.2em;
  --checkbox-control-border-color: var(--smokey-black);
  --checkbox-control-checked-bg-color: var(--smokey-black);

  /* Fieldset */
  --fieldset-background-color: var(--light-grey);
  --fieldset-border-radius: 1.5em;

  /* Dividers */

  --divider-color-primary: var(--smokey-black);
  --divider-background-color: var(--light-gray);
  --divider-border-color: var(--smokey-black);

  /* Home animation */

  --home-box-color: var(--silver-metallic);
  --home-grid-color: var(--light-grey);
}

:root.dark {
  /* Theme colours */

  --page-background-color: var(--jet);
  --background-color: var(--smokey-black);
  --background-color-alt: var(--granite-grey);
  --primary-color: var(--light-grey);
  --surface-color: var(--jet);

  --on-background-color: var(--light-grey);
  --on-primary-color: var(--smokey-black);
  --on-surface-color: var(--light-grey);

  /* Anchors */
  --anchor-color-on-surface: var(--accent-dark-900);

  /* Controls */

  --non-button-control-bg-color: #000;
  --non-button-control-text-color: var(--light-grey);

  --control-bg-color: var(--light-grey);
  --control-text-color: var(--smokey-black);
  --control-secondary-bg-color: var(--smokey-black);
  --control-secondary-text-color: var(--light-grey);
  --control-tertiary-border-color: var(--light-grey);
  --control-tertiary-text-color: var(--light-grey);

  /* Checkbox control */

  --checkbox-control-border-color: var(--light-grey);
  --checkbox-control-checked-bg-color: var(--light-grey);

  /* Fieldset */

  --fieldset-background-color: var(--smokey-black);

  /* Dividers */

  --divider-color-primary: var(--magnolia);
  --divider-background-color: var(--smokey-black);
  --divider-border-color: var(--magnolia);

  /* Home animation */

  --home-grid-color: var(--smokey-black);
}
