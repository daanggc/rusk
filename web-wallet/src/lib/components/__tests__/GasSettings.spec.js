import { afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, fireEvent, render } from "@testing-library/svelte";

import { GasSettings } from "..";
import { get } from "svelte/store";
import { settingsStore } from "$lib/stores";
import { createCurrencyFormatter } from "$lib/dusk/currency";

describe("GasSettings", () => {
  const settings = get(settingsStore);
  const duskFormatter = createCurrencyFormatter(settings.language, "DUSK", 9);
  const fee = duskFormatter(
    settings.gasPrice * settings.gasLimit * 0.000000001
  );

  const baseProps = {
    fee: fee,
    limit: 20000000,
    limitLower: 10000000,
    limitUpper: 1000000000,
    price: 1,
    priceLower: 1,
  };

  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("renders the GasSettings component closed", () => {
    const { container } = render(GasSettings, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the GasSettings component expanded", async () => {
    const { container, getByRole } = render(GasSettings, baseOptions);

    const next = getByRole("button", { name: "EDIT" });

    await fireEvent.click(next);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('checks "gasSettings" event is dispatched on click with the correct event data', async () => {
    const eventHandler = vi.fn();
    const { component, getByRole, getAllByRole } = render(
      GasSettings,
      baseOptions
    );
    const editButton = getByRole("button", { name: "EDIT" });

    expect(() => getAllByRole("spinbutton")).toThrow();

    await fireEvent.click(editButton);

    component.$on("gasSettings", eventHandler);

    const [priceInput, limitInput] = getAllByRole("spinbutton");

    await fireEvent.input(limitInput, {
      target: { value: baseProps.limitLower },
    });

    expect(eventHandler).toHaveBeenCalledTimes(1);
    expect(eventHandler.mock.lastCall?.[0].detail).toStrictEqual({
      limit: baseProps.limitLower,
      price: baseProps.price,
    });

    await fireEvent.input(priceInput, {
      target: { value: baseProps.price * 2 },
    });

    expect(eventHandler).toHaveBeenCalledTimes(2);
    expect(eventHandler.mock.lastCall?.[0].detail).toStrictEqual({
      limit: baseProps.limitLower,
      price: baseProps.price * 2,
    });
  });
});
