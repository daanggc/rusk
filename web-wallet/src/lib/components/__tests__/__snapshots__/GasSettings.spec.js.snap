// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GasSettings > renders the GasSettings component closed 1`] = `
<div
  class="gas-settings svelte-1daay07"
>
  <dl
    class="gas-settings__edit svelte-1daay07"
  >
    <div
      class="gas-fee svelte-1c6b481"
    >
      <span>
        Fee:
      </span>
       
      <div
        class="gas-fee__amount svelte-1c6b481"
      >
        <div
          class="gas-fee__amount-value svelte-1c6b481"
        >
          <span
            class="svelte-1c6b481"
          >
            max
          </span>
           
          <span
            class="svelte-1c6b481"
          >
            0.020000000
          </span>
        </div>
         
        <svg
          class="dusk-icon dusk-icon--size--default dusk-amount__icon"
          data-tooltip-id="main-tooltip"
          data-tooltip-text="DUSK"
          role="graphics-symbol"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.335 0.005a12.186 12.186 0 0 0 -2.58 0.204C4.2 1.259 0 6.14 0 11.999 0 17.859 4.203 22.74 9.759 23.79c0.726 0.138 1.476 0.21 2.241 0.21 6.789 0 12.261 -5.64 11.991 -12.49C23.741 5.249 18.599 0.173 12.335 0.005Zm0.207 21.58c-0.204 0.012 -0.336 -0.222 -0.21 -0.384 1.974 -2.541 3.153 -5.733 3.153 -9.202 0 -3.468 -1.176 -6.663 -3.153 -9.202 -0.126 -0.162 0.003 -0.396 0.207 -0.384 5.052 0.282 9.06 4.465 9.06 9.586 0 5.121 -4.005 9.304 -9.057 9.586Z"
          />
        </svg>
        
      </div>
    </div>
    <dd>
      <button
        class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--small"
        type="button"
      >
        <span
          class="dusk-button__text"
        >
          EDIT
        </span>
      </button>
    </dd>
  </dl>
   
</div>
`;

exports[`GasSettings > renders the GasSettings component expanded 1`] = `
<div
  class="gas-settings svelte-1daay07"
>
  <dl
    class="gas-settings__edit svelte-1daay07"
  >
    <div
      class="gas-fee svelte-1c6b481"
    >
      <span>
        Fee:
      </span>
       
      <div
        class="gas-fee__amount svelte-1c6b481"
      >
        <div
          class="gas-fee__amount-value svelte-1c6b481"
        >
          <span
            class="svelte-1c6b481"
          >
            max
          </span>
           
          <span
            class="svelte-1c6b481"
          >
            0.020000000
          </span>
        </div>
         
        <svg
          class="dusk-icon dusk-icon--size--default dusk-amount__icon"
          data-tooltip-id="main-tooltip"
          data-tooltip-text="DUSK"
          role="graphics-symbol"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.335 0.005a12.186 12.186 0 0 0 -2.58 0.204C4.2 1.259 0 6.14 0 11.999 0 17.859 4.203 22.74 9.759 23.79c0.726 0.138 1.476 0.21 2.241 0.21 6.789 0 12.261 -5.64 11.991 -12.49C23.741 5.249 18.599 0.173 12.335 0.005Zm0.207 21.58c-0.204 0.012 -0.336 -0.222 -0.21 -0.384 1.974 -2.541 3.153 -5.733 3.153 -9.202 0 -3.468 -1.176 -6.663 -3.153 -9.202 -0.126 -0.162 0.003 -0.396 0.207 -0.384 5.052 0.282 9.06 4.465 9.06 9.586 0 5.121 -4.005 9.304 -9.057 9.586Z"
          />
        </svg>
        
      </div>
    </div>
    <dd>
      <button
        class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--small"
        type="button"
      >
        <span
          class="dusk-button__text"
        >
          CLOSE
        </span>
      </button>
    </dd>
  </dl>
   
  <div
    class="gas-settings svelte-1daay07"
    style="animation: __svelte_276924520_0 400ms linear 0ms 1 both;"
  >
    <div
      class="dusk-exclusive-choice"
      role="radiogroup"
    >
      <input
        class="dusk-exclusive-choice__radio"
        id="2d814b13-5fe4-4036-b003-a4383070bebd-normal"
        name="2d814b13-5fe4-4036-b003-a4383070bebd"
        type="radio"
        value="normal"
      />
       
      <label
        class="dusk-exclusive-choice__label"
        for="2d814b13-5fe4-4036-b003-a4383070bebd-normal"
      >
        Normal
      </label>
      <input
        class="dusk-exclusive-choice__radio"
        id="2d814b13-5fe4-4036-b003-a4383070bebd-fast"
        name="2d814b13-5fe4-4036-b003-a4383070bebd"
        type="radio"
        value="fast"
      />
       
      <label
        class="dusk-exclusive-choice__label"
        for="2d814b13-5fe4-4036-b003-a4383070bebd-fast"
      >
        Fast
      </label>
    </div>
     
    <label
      class="gas-control svelte-14slcjg"
    >
      <span
        class="gas-control__label svelte-14slcjg"
      >
        Price: (lux)
      </span>
       
      <input
        class="dusk-textbox dusk-textbox-number gas-control__input"
        max="20000000"
        min="1"
        required=""
        type="number"
      />
      
    </label>
     
    <label
      class="gas-control svelte-14slcjg"
    >
      <span
        class="gas-control__label svelte-14slcjg"
      >
        Gas Limit: (unit)
      </span>
       
      <input
        class="dusk-textbox dusk-textbox-number gas-control__input"
        max="1000000000"
        min="10000000"
        required=""
        type="number"
      />
      
    </label>
  </div>
</div>
`;
